File: frontend/src/models/analyzer/value/TypeArgument.ts
import type { Type } from "@/models/analyzer/value/Type"
import type { TypeBound } from "@/models/analyzer/value/TypeBound"

export type TypeArgument = ConcreteTypeArgument | WildcardTypeArgument;

export interface ConcreteTypeArgument {
  type: Type;
}

export interface WildcardTypeArgument {
  wildcard: 'any' | TypeBound;
}

File: analyzer/src/main/java/com/mategka/dava/analyzer/struct/property/value/argument/TypeArgument.java
package com.mategka.dava.analyzer.struct.property.value.argument;

import com.fasterxml.jackson.annotation.JsonValue;

import java.io.Serializable;
import java.util.Map;

public sealed interface TypeArgument extends Serializable permits ConcreteTypeArgument, WildcardTypeArgument {

  @JsonValue
  Map<String, Object> toJsonValue();

}
